ARC: boards: ARCv3: add qemu_arc_hs5x board

evgeniy-paltsev opened the pull request:
Add qemu_arc_hs5x - single core ARCv3 32bit HS5x CPU.

NOTE: we don't enable zephyr toolchain for it as current Zephyr SDK doesn't have hs5x support. We'll enable it when we switch to 0.15.0 Zephyr SDK.

While I'm at it also do minor non-functional polishing for HS6x QEMU and update board documentation.

evgeniy-paltsev committed:
ARC: boards: ARCv3: add qemu_arc_hs5x board

Add qemu_arc_hs5x - single core ARCv3 32bit HS5x CPU.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>

ruuddw reviewed (commented):
See minor comments. Also, any documentation to be updated with this new board?
ARC support matrix has HS 5x already included, but not 6x. ARC QEMU board documentation I think is generic, nothing specific for 3x/4x/5x/6x, right?

evgeniy-paltsev commented:
Hi @ruuddw,

> ARC support matrix has HS 5x already included, but not 6x. 

Hmm, we have both hs6x and hs5x in the latest docs:
https://docs.zephyrproject.org/latest/hardware/arch/arc-support-status.html#support-status
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/hardware/arch/arc-support-status.rst#support-status

Though in the doc page there is horizontal scroll for the table - looks like it isn't really noticeable :/

evgeniy-paltsev committed:
ARC: boards: QEMU: hs6x: minor description polish

Non-functional polish of Kconfig symbol description and
QEMU_CPU_TYPE_${ARCH} variable (used only for verbose cmake comments)
for HS6x QEMU.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>

evgeniy-paltsev committed:
ARC: boards: QEMU: documentation update

Update ARC QEMU board documentation to reflect added
HS5x and HS6x support.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>

evgeniy-paltsev commented on boards/arc/qemu_arc/qemu_arc_hs5x.yaml:
```diff
@@ -0,0 +1,12 @@
+identifier: qemu_arc_hs5x
+name: QEMU Emulation for ARC HS
```
done, thanks!

evgeniy-paltsev commented on soc/arc/snps_qemu/Kconfig:
```diff
@@ -15,4 +15,7 @@ config SOC_QEMU_ARC_HS
 	bool "Synopsys ARC HS in QEMU"
 
+config SOC_QEMU_ARC_HS5X
+	bool "Synopsys ARC HS5x in QEMU"
```
done, thanks!

ruuddw reviewed (approved):
Thanks!
